src/ethereum.js

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect, useMemo } from "react";
+import { parseUnits } from "./helpers";
+import { ethers } from "ethers";
+import ERC20JSON from "@openzeppelin/contracts/build/contracts/ERC20";
+
+const { hexlify } = ethers.utils;
+
+export function useEthereumAccounts() {
+  const [accounts, setAccounts] = useState();
+  useEffect(async () => {
+    if (window.ethereum) {
```

Comment:
why do we have to store this info on the window? shortcut for now? this conditional of only running "setAccounts" when "accountsChanged" seems like something that could go in the useEffect second parameter that is currently [], except that since it's on the window, it's not part of the react world so it probably doesn't work, huh...

Reply:
Good eye!

This is part of the [MetaMask API](https://docs.metamask.io/guide/ethereum-provider.html#methods) which is added to window. You're right that using window is janky so the idea is to listen to events from `window.ethereum` and update them in React state accordingly.